### PR TITLE
Consistent delete/rename

### PR DESCRIPTION
**Pull request by jreidinger**
https://trello.com/c/9q8UEvBD/1326-5-network-handle-properly-removal-and-rename-of-interfaces
https://bugzilla.suse.com/show_bug.cgi?id=1153460

### CODE REVIEW

File: src/lib/y2network/config.rb [hunk 1]
```diff
@@ -205,6 +210,54 @@ def configured_interface?(iface_name)
       !connections.by_interface(iface_name).empty?
     end
 
+    # @note does not work recursivelly. So for delete it needs to be called for all modified vlans.
```
Comment (imobachgs) [inline src/lib/y2network/config.rb]: np: recursively

File: src/lib/y2network/config.rb [hunk 2]
```diff
@@ -143,7 +143,10 @@ def rename_interface(old_name, new_name, mechanism)
       interface = interfaces.by_name(old_name || new_name)
       interface.rename(new_name, mechanism)
       return unless old_name # do not modify configurations if it is just renaming mechanism
-      connections.by_interface(old_name).each { |c| c.interface = new_name }
+      connections.by_interface(old_name).each do |connection|
+        connection.interface = new_name
+        rename_dependencies(old_name, new_name, connection)
```
Comment (imobachgs) [inline src/lib/y2network/config.rb]: np: Perhaps you could omit the `old_name` or the `new_name` as it is contained in the `connection` instance.

Comment (jreidinger) [reply src/lib/y2network/config.rb]: where it is contained? I only found it in interface class, but not in connection one.

Comment (imobachgs) [reply src/lib/y2network/config.rb]: Is not `connection.interface`?

Comment (jreidinger) [reply src/lib/y2network/config.rb]: connection.interface is only the new name ( it is string ). So this way it is more explicit for me.

**Pull request merged**